data/hooks/diagnosis/14-ports.py

```diff
@@ -26,16 +26,20 @@ def run(self):
                 ports[port] = service
 
         try:
-            r = requests.post('https://diagnosis.yunohost.org/check-ports', json={'ports': ports.keys()}, timeout=30).json()
-            if "status" not in r.keys():
-                raise Exception("Bad syntax for response ? Raw json: %s" % str(r))
-            elif r["status"] == "error":
-                if "content" in r.keys():
-                    raise Exception(r["content"])
-                else:
+            r = requests.post('https://diagnosis.yunohost.org/check-ports', json={'ports': ports.keys()}, timeout=30)
+            if r.status_code == 200:
+                r = r.json()
+                if "status" not in r.keys():
                     raise Exception("Bad syntax for response ? Raw json: %s" % str(r))
-            elif r["status"] != "ok" or "ports" not in r.keys() or not isinstance(r["ports"], dict):
-                raise Exception("Bad syntax for response ? Raw json: %s" % str(r))
+                elif r["status"] == "error":
+                    if "content" in r.keys():
+                        raise Exception(r["content"])
+                    else:
+                        raise Exception("Bad syntax for response ? Raw json: %s" % str(r))
+                elif r["status"] != "ok" or "ports" not in r.keys() or not isinstance(r["ports"], dict):
+                    raise Exception("Bad syntax for response ? Raw json: %s" % str(r))
+            else:
+                raise Exception("Bad response from the server https://diagnosis.yunohost.org : %s" % str(r.status_code))
```

Comment:
You probably wants to put the full url with the path here, to ease diagnosis (and maybe display r.content in the error).

Same for after.

Reply:
Should be resolved by https://github.com/YunoHost/yunohost/pull/872/commits/4399c9b740b5c4901de76f5b2245df4b349eae89